test/e2e_federation/replicaset.go

```diff
@@ -462,7 +461,20 @@ func newReplicaSet(namespace string, prefix string, replicas int32, pref *federa
 		prefString := string(prefBytes)
 		rs.Annotations[fedreplicsetcontroller.FedReplicaSetPreferencesAnnotation] = prefString
 	}
-	return &rs
+	return rs
+
+}
+
+func newReplicaSet(namespace string, prefix string, replicas int32, pref *federation.FederatedReplicaSetPreferences) *v1beta1.ReplicaSet {
+	rs := newReplicaSetObj(namespace, replicas, pref)
+	rs.GenerateName = prefix
+	return rs
+}
+
+func newReplicaSetWithName(namespace string, name string, replicas int32, pref *federation.FederatedReplicaSetPreferences) *v1beta1.ReplicaSet {
+	rs := newReplicaSetObj(namespace, replicas, pref)
+	rs.Name = name
+	return rs
 }
 
 func extraceClusterNames(clusters map[string]*cluster) []string {
```

Comment:
L120: I think you want rs = createReplicaSetOrFail(f.FederationClientset, rs) or else you have the wrong object in the cleanup function that follows.